### PR TITLE
cpu/efm32: cleanup

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

`efm32` uses shell calls to resolve a series of `CPU` parameters from the `CPU_MODEL`. All these shell calls are slow and make building and testing any `efm32` slower than it needs to be.

This PR fixes this by aggregating all that data into one single file and using `MAKE` functions to extract the data.

A benchmark example is that building and running all tests without this PR takes about 1:49 on my machine, and it now takes 1:15.

This also move those variables to Makefile.include or Makefile.features respectively according to where they should be.

Names are changed to use the already existing and exported `CPU_ARCH` and `CPU_FAM`.

### Testing procedure

Look a t the number of execv calls in this PR vs master:

- master

```
BOARD=slstk3402a strace -ff -e trace=execve make --no-print-directory -C examples/hello-world/ info-debug-variable-QUIET 2>&1 | grep 'execve' | wc -l
5571
```

- pr

```
BOARD=slstk3402a strace -ff -e trace=execve make --no-print-directory -C examples/hello-world/ info-debug-variable-QUIET 2>&1 | grep 'execve' | wc -l
83
```

You can also compare time used to resolve simple make targets e.g.:

- master

```
time BOARD=slstk3402a make --no-print-directory -C examples/hello-world/ info-debug-variable-USEMODULE
auto_init board boards_common_silabs core core_msg cortexm_common cortexm_common_periph cortexm_fpu cpu cpu_efm32pg12b gecko_sdk_emlib gecko_sdk_emlib_extra newlib newlib_nano newlib_syscalls_default periph periph_common periph_gpio periph_pm periph_uart pm_layered silabs_aem silabs_bc stdio_uart sys

real    0m0.741s
user    0m0.555s
sys     0m0.193s
```

-pr

```
time BOARD=slstk3402a make --no-print-directory -C examples/hello-world/ info-debug-variable-USEMODULE
auto_init board boards_common_silabs core core_msg cortexm_common cortexm_common_periph cortexm_fpu cpu cpu_efm32pg12b gecko_sdk_emlib gecko_sdk_emlib_extra newlib newlib_nano newlib_syscalls_default periph periph_common periph_gpio periph_pm periph_uart pm_layered silabs_aem silabs_bc stdio_uart sys

real    0m0.170s
user    0m0.115s
sys     0m0.027s
```

### Issues/PRs references
 
Relates to #10850
Also relates to #9913
This is a first cleanup step, when #12407 is merged and agreed I will move towards a folder based approach.

### CODE REVIEW

**Comment by basilfx** [comment]
That's a very nice refactoring I never thought of. It's still very maintainable.

I have to test this in detail, hope to do that this week.

File: cpu/efm32/Makefile.include [hunk 1]
```diff
@@ -18,16 +20,16 @@ ifeq ($(CPU_ARCH),cortex-m0plus)
 endif
 
 # include CPU family module
-USEMODULE += cpu_$(EFM32_FAMILY)
+USEMODULE += cpu_$(CPU_FAM)
```
Comment (aabadie) [inline cpu/efm32/Makefile.include]: Could this be moved in a extra Makefile.dep file ?
(same for the pm_layered module below in this file)

File: cpu/efm32/efm32-features.mk [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+
+include $(RIOTCPU)/efm32/efm32-vars.mk
+
+CPU_FAM      = $(word $(CPU_FAM_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
+CPU_SERIES   = $(word $(CPU_SERIES_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
+CPU_ARCH     = $(word $(CPU_ARCH_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
+
+EFM32_CRYPTO = $(word $(EFM32_CRYPTO_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
+EFM32_TRNG   = $(word $(EFM32_TRNG_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
+EFM32_RADIO  = $(word $(EFM32_RADIO_INDEX), $(EFM32_VARS_$(CPU_MODEL)))
```
Comment (aabadie) [inline cpu/efm32/efm32-features.mk]: missing newline.

File: cpu/efm32/efm32-info.mk [hunk 3]
```diff
@@ -13,15 +13,15 @@ ifeq (,$(EFM32_INFO))
...
-export EFM32_ARCHITECTURE = $(word 4, $(EFM32_INFO))
+CPU_FAM = $(word 2, $(EFM32_INFO))
+CPU_SERIES = $(word 3, $(EFM32_INFO))
+CPU_ARCH = $(word 4, $(EFM32_INFO))
 
-export EFM32_FLASH_START = $(word 5, $(EFM32_INFO))
-export EFM32_FLASH_SIZE = $(word 6, $(EFM32_INFO))
-export EFM32_SRAM_START = $(word 7, $(EFM32_INFO))
-export EFM32_SRAM_SIZE = $(word 8, $(EFM32_INFO))
+EFM32_FLASH_START = $(word 5, $(EFM32_INFO))
```
Comment (aabadie) [inline cpu/efm32/efm32-info.mk]: This variable is already set [here](https://github.com/RIOT-OS/RIOT/pull/13174/files#diff-22687e9fbc629a5d3b242f1cb23315b2R3) but with a different value (apparently). Which one is correct ?
(same question for the 3 other variables below).

File: cpu/efm32/efm32-features.mk (hunk 2 above)
Comment (basilfx) [reply cpu/efm32/efm32-features.mk]: I also think that the variables are not needed in this file.

Comment (fjmolinas) [reply cpu/efm32/efm32-features.mk]: They are not used but I kept them, should I remove them al together?

File: cpu/efm32/Makefile.include (hunk 1 above)
Comment (fjmolinas) [reply cpu/efm32/Makefile.include]: OK!

File: cpu/efm32/efm32-info.mk (hunk 3 above)
Comment (fjmolinas) [reply cpu/efm32/efm32-info.mk]: All the variables here should be removed, I missed that when rebasing I think.

File: cpu/efm32/Makefile [hunk 4]
```diff
@@ -1,6 +1,8 @@
 MODULE = cpu
 
 # add a list of subdirectories, that should also be build
-DIRS = periph families/$(EFM32_FAMILY) $(RIOTCPU)/cortexm_common
+DIRS += periph
+DIRS += families/$(CPU_FAM)
```
Comment (basilfx) [inline cpu/efm32/Makefile]: I recall a discussion where I was misusing `CPU_FAM` for this purpose, and had renamed it to `EFM32_FAMILY`. I guess that changed?

Comment (fjmolinas) [reply cpu/efm32/Makefile]: Oh, thats why it was there. I'll change it back since it is indeed not the right use.

File: cpu/efm32/efm32-features.mk (hunk 2 above)
Comment (basilfx) [reply cpu/efm32/efm32-features.mk]: Eventually, this should be removed when moving to KConfig, so not needed here.

File: cpu/efm32/Makefile (hunk 4 above)
Comment (fjmolinas) [reply cpu/efm32/Makefile]: kept the old variable and exported it. Maybe we could move them all to separate dirst like `stm32`, not sure what is best.

File: cpu/efm32/efm32-info.mk (hunk 3 above)
Comment (fjmolinas) [reply cpu/efm32/efm32-info.mk]: done.

File: cpu/efm32/efm32-features.mk (hunk 2 above)
Comment (fjmolinas) [reply cpu/efm32/efm32-features.mk]: removed

File: cpu/efm32/Makefile.include (hunk 1 above)
Comment (fjmolinas) [reply cpu/efm32/Makefile.include]: moved

File: cpu/efm32/efm32-vars.mk [hunk 5]
```diff
@@ -0,0 +1,139 @@
+CPU_FAM_INDEX           = 1
```
Comment (fjmolinas) [inline cpu/efm32/efm32-vars.mk]: BTW @basilfx I didn't keep the message that was in `cpu.txt`:

```
# This file is automatically generated, and should not be changed. There is
# probably little reason to edit this file anyway, since it should already
# contain all information for the EFM32GG family of CPUs.
```

since this wasn't generated automatically but I did copy paste everything so not data was changed.

**Comment by fjmolinas** [comment]
Rebased and fixed typo.

**Comment by aabadie** [comment]
There are changes related to kinetis, but unrelated to this PR and stk3700 builds are failing (there might be others)

**Comment by fjmolinas** [comment]
@aabadie murdock is passing now.

**Comment by fjmolinas** [comment]
@aabadie @basilfx could you give a couple of boards a test?

**Comment by basilfx** [comment]
Sorry for not being that active these weeks. I have other work that has priority.

I do have a few comments.

- `efm32-info.mk` isn't used anymore. That file was responsible for parsing the values, and provide meaningful errors (there are more supported CPU's in the 'database' than we have includes for).
- `efm32-features.mk` will be removed once we move on to Kconfig (#13111).
- I liked the idea of having the CPU-related information per family, instead of having them all in one file (`efm32-vars.mk`). That could be realized by defining CPU_FAM upfront in the board.
- We should not use `CPU_SERIES`: it's an EFM32 thingy.

@fjmolinas I really like the improvements, and I want this to be merged. But let me check if I can propose some changes (I'll do that right away).

**Comment by basilfx** [comment]
@fjmolinas I have modified a few things that I think keeps things more self-contained. I have three additional commit in my own branch [here](https://github.com/basilfx/RIOT/tree/pr_efm32_cleanup).

Here is the summary:

- Define `CPU_FAM` per board. That variable is allowed/provided, according to [here](https://github.com/RIOT-OS/RIOT/blob/master/makefiles/vars.inc.mk#L16).
  - Having `CPU_FAM` defined, it saves the shell execution [here](https://github.com/RIOT-OS/RIOT/blob/master/cpu/efm32/efm32-info.mk#L9), which you already fixed!
- Remove `CPU_SERIES`, as that is something that is confusing next to `CPU_FAM` and `CPU_ARCH`. Now that `CPU_FAM` is provided, all other references to `EFM32_FAMILY` can be removed.
- Restore families being self-contained, which makes it easy (for me :-P) to drop-in/symlink additional families, for testing.
- Restore the assertions in `efm32-info.mk` to provide meaningful errors in case headers or family-specific information is missing. 
  - Only headers for `CPU_MODEL`s that are used by a board are included, but the list of family-specific information includes all information about the CPUs.
  - To support another EFM32 `CPU_MODEL`, it should be as simple adding the new header file only. I use that for my custom boards.
  - And yes, by switching branches a lot, you sometimes forget that your board isn't supported in that branch :-P
- Keep the logic of parsing family-specific information central at one place and only use the `EFM32_` variables throughout the EFM32 CPU module only. This removes the need for the index variables, and keeps that logic at one place.
  - Only `EFM32_ARCHITECTURE` is used to provide `CPU_ARCH`.
  - No exports needed anymore.
- Fix support for `periph_hwrng` by providing `EFM32_TRNG` (and fix my typo).
- Remove logic from `efm32-features.mk` that will be removed eventually.
  - I think I chose a misleading file name, as it should only contain toggles to remove/fine-tune/experiment EFM32 related features.

Using the `strace` command, I'm at 59 `execve` syscalls (for `hello-world`), which is exactly the same as your branch. Compared to master, it is really much faster! I never thought of this issue.

I want to stress out that I don't want to annoy you by fixing some things myself, after you and @aabadie have almost finished this. But I hope you can consider my changes. You did the hard work, I only find-replaced/moved some parts.

**Comment by fjmolinas** [comment]
Hi @basilfx thanks for your feedback, some comments

- `CPU_ARCH ` can't be defined in `Makefile.include`, these goes against #9913, but it can be moved to `Makefile.features`

> Remove logic from efm32-features.mk that will be removed eventually. 

I don't mind removing this.

```diff
- DIRS += families/$(EFM32_FAMILY)
+ DIRS += families/$(CPU_FAM)
```

This shouldn't be done, because `CPU_FAM` is only available here because it is exported, but the is no real reason to export is, at some point that export should be removed, so I rather use an explicit variable `EFM32_FAMILY` that is exported just for `efm32`. You were the one that mentioned in a previous comment that `CPU_FAM` should not be used this way. Not using it for `VECTORS_O := $(BINDIR)/cpu_$(CPU_FAM)/vectors.o` and others related is for the same reason. 

If you can recall or find that discussion it would be great, to either make clear that `CPU_FAM` can be use that way or move away from that kind of usage.

> Restore families being self-contained, which makes it easy to drop-in/symlink additional families, for testing.

As I said in  the PR eventually we will have to move towards https://github.com/RIOT-OS/RIOT/pull/12407, which will mean one directory per `CPU_MODEL`, but if you prefer keeping these separate for now I don't mind. But if https://github.com/RIOT-OS/RIOT/pull/12407 is implemented they will disappear anyway.

**Comment by fjmolinas** [comment]
> If you can recall or find that discussion it would be great, to either make clear that CPU_FAM can be use that way or move away from that kind of usage.

I'm guessing its probably a comment made by @cladmi  no?

**Comment by basilfx** [comment]
@fjmolinas I've send you an email to maybe discuss/explain this ;-)

**Comment by fjmolinas** [comment]
Talking offline with @basilfx and since `CPU_FAM` is already used for directory resolution (which is why it needs to be exported). I'll go ahead with @basilfx suggestion and use `CPU_FAM` as well.

@basilfx can we remove the column with the CPU_FAM from the "database" then?

**Comment by fjmolinas** [comment]
> @basilfx can we remove the column with the CPU_FAM from the "database" then?

Maybe it is usefull for new boards?

**Comment by cladmi** [comment]
>> If you can recall or find that discussion it would be great, to either make clear that CPU_FAM can be use that way or move away from that kind of usage.

> I'm guessing its probably a comment made by @cladmi no?

> Talking offline with @basilfx and since `CPU_FAM` is already used for directory resolution (which is why it needs to be exported). I'll go ahead with @basilfx suggestion and use `CPU_FAM` as well.

TL;DR; I would not have exported either `CPU_FAM` or `EFM32_FAMILY` but used different module names for each family and done the `DIRS` resolution based on `USEMODULE`.

---- 

The reason I would not have exported `CPU_FAM`, but also not `EFM32_FAMILY`, for this use case, is that the reason to go into that `families/cpu_something` directory is not because of this cpu is from this `CPU_FAM` but because the module in that directory is used.
Currently they may all have the same module name `cpu` or `cpu_common` (no idea), so a hack is used to find where the implementation is, but if being explicit and renaming it to different names `cpu_efm32_family_CPU_FAM`, then it would not matter in the rest of the building process that it is `CPU_FAM`, or `CPU_LINE`, or `CPU_REVISION`, the module name would be sufficient.

Then adding the `families/cpu_something` to `DIRS` would not depend of `CPU_FAM` but on `USEMODULE`. Which for me matched way more a declarative model where each module has a known directory. I was in favor of replacing `board` and `cpu` by `board_BOARD_NAME` and `cpu_CPU_NAME`.

With this being declarative, and moving the definition earlier in the build, it could allow moving this resolution in the main `Makefile.include`, instead of in many recursive `Makefile`, and so let `make` figure out how to parallelize the compilation of each module and maybe not even go in a module directory if it is up to date.

**Comment by basilfx** [comment]
> @basilfx can we remove the column with the CPU_FAM from the "database" then?

I've updated the PR to remove that column as well. Makes sense!

> TL;DR; I would not have exported either `CPU_FAM` or `EFM32_FAMILY` but used different module names for each family and done the `DIRS` resolution based on `USEMODULE`.

I guess that could be a next improvement? I see that this one already saves a lot, and the other approach would also include work outside of the EFM32 CPU folder, right?

**Comment by cladmi** [comment]
> > TL;DR; I would not have exported either `CPU_FAM` or `EFM32_FAMILY` but used different module names for each family and done the `DIRS` resolution based on `USEMODULE`.
> 
> I guess that could be a next improvement? I see that this one already saves a lot, and the other approach would also include work outside of the EFM32 CPU folder, right?

Exactly. I only answered to the why not `export` question.
But in the current situation, either `EFM32_FAMILY` or now `CPU_FAM` needs to be exported.

**Comment by kaspar030** [comment]
The improvement in build time is *huge*.
On my Thinkpad, this PR:
```
[kaspar@ng riot.tmp/examples/hello-world (pr13174)]$ hyperfine "BOARDS=\"ikea-tradfri slstk3401a slstk3402a sltb001a stk3600 stk3700\" make buildtest"
Benchmark #1: BOARDS="ikea-tradfri slstk3401a slstk3402a sltb001a stk3600 stk3700" make buildtest
  Time (mean ± σ):     15.885 s ±  0.509 s    [User: 17.479 s, System: 6.611 s]
  Range (min … max):   15.289 s … 17.022 s    10 runs
```

master:
```
[kaspar@ng riot.tmp/examples/hello-world (master)]$ hyperfine "BOARDS=\"ikea-tradfri slstk3401a slstk3402a sltb001a stk3600 stk3700\" make buildtest"
Benchmark #1: BOARDS="ikea-tradfri slstk3401a slstk3402a sltb001a stk3600 stk3700" make buildtest
  Time (mean ± σ):     34.981 s ±  0.663 s    [User: 33.810 s, System: 10.956 s]
  Range (min … max):   33.748 s … 36.087 s    10 runs
```

@basilfx @cladmi I'm inclined to ACK this. Are you OK with the current state?

**Comment by fjmolinas** [comment]
> @basilfx @cladmi I'm inclined to ACK this. Are you OK with the current state?

let me address @basilfx suggestions first.

File: cpu/efm32/efm32-info.mk [hunk 6]
```diff
@@ -1,13 +1,27 @@
...
+EFM32_ARCHITECTURE = $(word 2, $(EFM32_INFO))
+
+EFM32_FLASH_START  = $(word 3, $(EFM32_INFO))
+EFM32_FLASH_SIZE   = $(word 4, $(EFM32_INFO))
+EFM32_SRAM_START   = $(word 5, $(EFM32_INFO))
+EFM32_SRAM_SIZE    = $(word 6, $(EFM32_INFO))
+
+EFM32_CRYPTO       = $(word 7, $(EFM32_INFO))
+EFM32_TRNG         = $(word 8, $(EFM32_INFO))
+EFM32_RADIO        = $(word 9, $(EFM32_INFO))
```
Comment (kaspar030) [inline cpu/efm32/efm32-info.mk]: missing newline

**Comment by fjmolinas** [comment]
@basilfx @kaspar030 @aabadie I pushed changes implemeting multiple things:

- remove need of exports for `CPU_FAM`, use a `MODULE` logic instead
- avoid using `CPU_FAM` for directory resolution except in cases where:
    1 `CPU_FAM` is available without export
    2 `CPU_FAM` is used to include variables
    3 can't be done because of #9913

- I restored the family containment as requested by @basilfx, defined `CPU_FAM` per `BOARD` although probably not needed. Can be removed later.

- fixed the `EFM32_TRNG` typo
- not use `CPU_SERIES`
- Move all include related logic to the `families` `MODULE` `Makefile.include.`

**Comment by kaspar030** [comment]
This is a rather large PR. It *looks* good to me.
I tried some compiling, also after the latest changes it is as much faster as before.

I'm not sure how to continue. I have a sltb001a which I could use to do a full test run.

**Comment by fjmolinas** [comment]
> This is a rather large PR. It looks good to me.
> I tried some compiling, also after the latest changes it is as much faster as before.

i did a compile test so I think that at least should be OK. I have an efm32 I'll do a run on one of those as well.

**Comment by fjmolinas** [comment]
> i did a compile test so I think that at least should be OK.

A simple one, one for each board we have only.

**Comment by kaspar030** [comment]
IMO you can squash this.

**Comment by kaspar030** [comment]
CI is only unhappy about the non-squashed state.

**Comment by kaspar030** [comment]
My buildtest run is somewhere at "e" (event_wait_timeout) :(

**Comment by fjmolinas** [comment]
This where my run results, excluding from tests:

`--applications-exclude="tests/gnrc_ipv6_ext tests/gnrc_ipv6_ext_frag tests/gnrc_rpl_srh tests/gnrc_dhcpv6_client tests/gnrc_sock_dns tests/gnrc_tcp tests/driver_grove_ledbar tests/driver_my922 tests/driver_ds1307 tests/pkg_fatfs_vfs"`

```
ERROR:slstk3402a:Tests failed: 5
Failures during test:
- [examples/micropython](examples/micropython/test.failed)
- [examples/suit_update](examples/suit_update/test.failed)
- [tests/periph_gpio](tests/periph_gpio/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/sys_arduino](tests/sys_arduino/test.failed)
```

**Comment by fjmolinas** [comment]
> My buildtest run is somewhere at "e" (event_wait_timeout) :(

can you explain?

**Comment by kaspar030** [comment]
> can you explain?

it takes ages. :)

**Comment by fjmolinas** [comment]
> - [tests/periph_gpio](tests/periph_gpio/test.failed)

This is because it set `uart` gpios. Other failures are expected or already in master.

```
2020-02-11 17:54:37,746 #  bench 0 5
2020-02-11 17:54:37,746 # 
2020-02-11 17:54:37,748 # GPIO driver run-time performance benchmark
2020-02-11 17:54:37,748 # 
2020-02-11 17:54:37,933 #                  nop loop:    175002us  ---   0.175us per call  ---    5714220 calls per sec
2020-02-11 17:54:38,490 #                  gpio_set:    550002us  ---   0.550us per call  ---    1818175 calls per sec
2020-02-11 17:54:40,901 # Y.���� ---    1481477 calls                gpio_write:    650003us  ---   0.650us per call  ---    1538454 calls per sec
2020-02-11 17:54:40,901 # 
2020-02-11 17:54:40,902 #  --- DONE ---
```

**Comment by fjmolinas** [comment]
This became a nightmare to rebase...

**Comment by fjmolinas** [comment]
I squashed and re-wrote the commit history  to make it cleaner, it was compiling on every commit. I rebased to make sure I wasn't making any changes with respect to the past history, the only changes were to remove unneeded `:=` and move an error message location. With the new commit history I think it should be easier to review individual changes.

If name-spacing is not wanted for the `famlies/cpu_*` modules I can rename them pack to just the family. I though it would be good to namespace to not mixup the module and the `CPU_FAM` variable.

**Comment by basilfx** [comment]
I prefer to have the folder in `families/` as they were. Similar to the folders in `cpu/`, they are not prefixed with `cpu_` either, and most of the files in `families/<name>/` are vendor files to get the `cpu/efm32/` working.

**Comment by fjmolinas** [comment]
> I prefer to have the folder in families/ as they were. Similar to the folders in cpu/, they are not prefixed with cpu_ either, and most of the files in families/<name>/ are vendor files to get the cpu/efm32/ working.

Ok, I'll se when I have time to do this. Probably not soon.

**Comment by fjmolinas** [comment]
> Ok, I'll se when I have time to do this. Probably not soon.

Well did it right away, I think you will he happy with what is now @basilfx

File: cpu/efm32/Makefile [hunk 7]
```diff
@@ -1,6 +1,26 @@
 MODULE = cpu
 
 # add a list of subdirectories, that should also be build
-DIRS = periph families/$(EFM32_FAMILY) $(RIOTCPU)/cortexm_common
+DIRS += periph
+DIRS += $(RIOTCPU)/cortexm_common
+
+ifneq (,$(filter cpu_efm32gg,$(USEMODULE)))
+	DIRS += families/efm32gg
```
Comment (aabadie) [inline cpu/efm32/Makefile]:
```suggestion
  DIRS += families/efm32gg
```
here and below (seems like there are tabs instead of 2 spaces)

**Comment by aabadie** [comment]
The CI is green, @basilfx are you ok with the current state of this PR ?

**Comment by kaspar030** [comment]
ACK from my side, but please @basilfx take a final look!

**Comment by basilfx** [comment]
I approve. Thanks for the hard work @fjmolinas!

**Comment by cladmi** [comment]
GG

File: cpu/efm32/families/efm32gg/efm32-info.mk [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# This file is automatically generated, and should not be changed. There is
+# probably little reason to edit this file anyway, since it should already
+# contain all information for the EFM32GG family of CPUs.
```
Comment (benpicco) [inline cpu/efm32/families/efm32gg/efm32-info.mk:3]: I'm trying to add support to another member of the efm32 family and now I wonder what tool was used to generate that kind of file.

Comment (basilfx) [reply cpu/efm32/families/efm32gg/efm32-info.mk:3]: You might want to look here: https://github.com/basilfx/EFM2RIOT/tree/develop/dist

It should be reasonably up to date and contain all supported chips already.

Comment (benpicco) [reply cpu/efm32/families/efm32gg/efm32-info.mk:3]: Uh that's nice.
@aabadie did something very similar for stm32.

I take we can't do that on the fly as Gecko SDK is required. But I don't like copying around folders either.

But we could just import that wholesale.

Comment (basilfx) [reply cpu/efm32/families/efm32gg/efm32-info.mk:3]: I don't think it is necessary to do it on-the-fly. These families are quite stable (only new ones are added).

Comment (benpicco) [reply cpu/efm32/families/efm32gg/efm32-info.mk:3]: So how about we import https://github.com/basilfx/EFM2RIOT/tree/develop/dist/cpu/efm32/families into RIOT as a whole?

File: cpu/efm32/families/efm32gg/efm32-info.mk [hunk 9]
```diff
@@ -0,0 +1,47 @@
+# This file is automatically generated, and should not be changed. There is
```
Comment (gschorcht) [inline cpu/efm32/families/efm32gg/efm32-info.mk:1]: @basilfx I have a THUNDERBOARD EFM32GG12 here, for whose MCU there is no entry in this file. Is this file really created automatically and if so, how. Or does the entry have to be added manually?

Comment (basilfx) [reply cpu/efm32/families/efm32gg/efm32-info.mk:1]: This file is generated using https://github.com/basilfx/EFM2Riot. That script parses the Gecko SDK and generates the necessary files. I'll admit that it's a bit outdated now.
